dali/kernels/imgproc/pointwise/multiply_add.h

```diff
@@ -16,24 +16,57 @@
 #define DALI_KERNELS_IMGPROC_POINTWISE_MULTIPLY_ADD_H_
 
 #include <utility>
-#include "dali/util/ocv.h"
 #include "dali/core/common.h"
 #include "dali/core/convert.h"
-#include "dali/core/geom/box.h"
 #include "dali/core/error_handling.h"
-#include "dali/kernels/kernel.h"
+#include "dali/core/geom/box.h"
 #include "dali/kernels/imgproc/roi.h"
+#include "dali/kernels/kernel.h"
+#include "dali/util/ocv.h"
 
 namespace dali {
 namespace kernels {
 
-template<typename OutputType, typename InputType, int ndims = 3>
+template <typename Out, typename In>
+struct MultiplyAddElementCpu {
+  MultiplyAddElementCpu(float addend, float multiplier)
+      : addend_{addend}, multiplier_{multiplier} {}
+
+  Out operator()(In element) {
+    return ConvertSat<Out>(element * multiplier_ + addend_);
+  }
+
+  float addend_;
+  float multiplier_;
+};
+
+template <typename Out>
+struct MultiplyAddElementCpu<Out, uint8_t> {
```

Comment:
You can also handle `int8_t` and `bool`.
With int8_t, you can reinterpret to `uint8_t`.

Reply:
done